src/api/utils/notifications/email_notices.py

```diff
@@ -46,29 +47,43 @@ def send_email(**kwargs):
     :kwarg recipients:
     :return bool:
     """
-    mail_gun_url = current_app.config["MAIL_GUN_URL"]
-    mail_gun_api_key = current_app.config["MAIL_GUN_API_KEY"]
-    mail_gun_test = current_app.config.get("MAIL_GUN_TEST")
-
-    if kwargs["sender"] == "":
+    if not kwargs["sender"] or not kwargs["sender"].strip():
         raise ValueError("sender address missing")
 
-    if mail_gun_test:
-        if validate_email(kwargs["recipients"]):
-            return True
+    validate_email(kwargs["recipients"])
 
-    if validate_email(kwargs["recipients"]):
-        response = requests.post(
-            mail_gun_url,
-            auth=("api", mail_gun_api_key),
-            data={
-                "from": kwargs["sender"],
-                "to": kwargs["recipients"],
-                "subject": kwargs["subject"],
-                "text": kwargs["message"]
-            }
-        )
-        if response.status_code == 200:
+    if os.getenv("MAIL_GUN_TEST"):
             return True
-        else:
-            return False
+
+    mail_gun_url = flask_app.config["MAIL_GUN_URL"]
+    mail_gun_api_key = flask_app.config["MAIL_GUN_API_KEY"]
+
+    data = {
+        "from": kwargs["sender"],
+        "to": kwargs["recipients"],
+        "subject": kwargs["subject"],
+        "text": kwargs["message"],
+    }
+    if kwargs.get("html"):
+        data.update(dict(html=kwargs.get("html")))
+    response = requests.post(
+        mail_gun_url, auth=("api", mail_gun_api_key), data=data
+    )
+    if response.status_code == 200:
+        return True
+    else:
+        return False
```

Comment:
Celery has a retry method for its tasks that might be useful here. Worth noting in the next PR.